fe/fe-core/src/main/java/com/starrocks/external/iceberg/ExpressionConverter.java

```diff
@@ -4,111 +4,168 @@
             return null;
         }
 
-        if (!(predicate.getChild(0) instanceof SlotRef)) {
+        if (columnExpr instanceof SlotRef) {
+            colName = ((SlotRef) columnExpr).getDesc().getColumn().getName();
+        } else if (columnExpr instanceof CastExpr) {
+            CastExpr columnCastExpr = (CastExpr) columnExpr;
+            colName = ((SlotRef) columnCastExpr.getChild(0)).getDesc().getColumn().getName();
+        } else {
             return null;
         }
-        SlotRef ref = (SlotRef) predicate.getChild(0);
 
-        if (!(predicate.getChild(1) instanceof LiteralExpr)) {
-            return null;
+        if (expr instanceof IsNullPredicate) {
+            IsNullPredicate isNullPredicate = (IsNullPredicate) expr;
+            if (isNullPredicate.isNotNull()) {
+                return notNull(colName);
+            } else {
+                return isNull(colName);
+            }
         }
-        LiteralExpr literal = (LiteralExpr) predicate.getChild(1);
 
-        String colName = ref.getDesc().getColumn().getName();
-        UnboundPredicate unboundPredicate = null;
-        switch (literal.getType().getPrimitiveType()) {
-            case BOOLEAN: {
-                unboundPredicate = Expressions.predicate(op, colName,
-                        ((BoolLiteral) literal).getValue());
-                break;
+        if (expr instanceof BinaryPredicate) {
+            BinaryPredicate binaryPredicate = (BinaryPredicate) expr;
+            LiteralExpr literal = (LiteralExpr) expr.getChild(1);
+            Object value = convertLiteral(literal);
+            if (value == null) {
+                return null;
             }
-            case TINYINT:
-            case SMALLINT:
-            case INT: {
-                unboundPredicate = Expressions.predicate(op, colName,
-                        (int) ((IntLiteral) literal).getValue());
-                break;
+            switch (binaryPredicate.getOp()) {
+                case LT:
+                    return lessThan(colName, value);
+                case LE:
+                    return lessThanOrEqual(colName, value);
+                case GT:
+                    return greaterThan(colName, value);
+                case GE:
+                    return greaterThanOrEqual(colName, value);
+                case EQ:
+                    return equal(colName, value);
+                case NE:
+                    return notEqual(colName, value);
+                default:
+                    return null;
+            }
+        }
+
+        if (expr instanceof InPredicate) {
+            InPredicate inPredicate = (InPredicate) expr;
+            List<Expr> valueExprList = inPredicate.getListChildren();
+            List<Object> values = new ArrayList<>(valueExprList.size());
+            for (Expr valueExpr : valueExprList) {
+                LiteralExpr literalExpr = (LiteralExpr) valueExpr;
+                values.add(convertLiteral(literalExpr));
```

Comment:
we may add null into list here
```suggestion
                values.add(convertLiteral(literalExpr));
```